nomad/leader.go

```diff
@@ -392,9 +402,16 @@ func (s *Server) reapFailedEvaluations(stopCh chan struct{}) {
 			newEval.StatusDescription = fmt.Sprintf("evaluation reached delivery limit (%d)", s.config.EvalDeliveryLimit)
 			s.logger.Printf("[WARN] nomad: eval %#v reached delivery limit, marking as failed", newEval)
 
+			// Create a follow-up evaluation that will be used to retry the
+			// scheduling for the job after the cluster is hopefully more stable
+			// due to the fairly large backoff.
+			followupEvalWait := failedEvalFollowUpBaseLineWait +
+				time.Duration(rand.Int63n(int64(failedEvalFollowUpWaitRange)))
+			followupEval := eval.CreateFailedFollowUpEval(followupEvalWait)
+
 			// Update via Raft
 			req := structs.EvalUpdateRequest{
-				Evals: []*structs.Evaluation{newEval},
+				Evals: []*structs.Evaluation{newEval, followupEval},
```

Comment:
Can you rename `newEval` to `updateEval`. It took me a while to realize it's not a "new" eval, just an update of the existing one.